setup.sh

```diff
@@ -601,53 +602,116 @@ if [ "$noperlpath" = "" ]; then
-
-echo "#!/bin/sh" >>$config_dir/restart
-echo "$config_dir/stop --kill && $config_dir/start" >>$config_dir/restart
-
-echo "#!/bin/sh" >>$config_dir/reload
-echo "echo Reloading Webmin server in $wadir" >>$config_dir/reload
-echo "pidfile=\`grep \"^pidfile=\" $config_dir/miniserv.conf | sed -e 's/pidfile=//g'\`" >>$config_dir/reload
-echo "kill -USR1 \`cat \$pidfile\`" >>$config_dir/reload
-
-chmod 755 $config_dir/start $config_dir/stop $config_dir/restart $config_dir/reload
+	echo "exec '$wadir/miniserv.pl' \$* $config_dir/miniserv.conf" >>$config_dir/start-init
+fi
+# Stop main
+echo "#!/bin/sh" >>$config_dir/stop-init
+echo "if [ \"\$1\" = \"--kill\" ]; then" >>$config_dir/stop-init
+echo "  echo Force stopping Webmin server in $wadir" >>$config_dir/stop-init
+echo "else" >>$config_dir/stop-init
+echo "  echo Stopping Webmin server in $wadir" >>$config_dir/stop-init
+echo "fi" >>$config_dir/stop-init
+echo "pidfile=\`grep \"^pidfile=\" $config_dir/miniserv.conf | sed -e 's/pidfile=//g'\`" >>$config_dir/stop-init
+echo "pid=\`cat \$pidfile\` 2>/dev/null" >>$config_dir/stop-init
+echo "if [ \"\$pid\" != \"\" ]; then" >>$config_dir/stop-init
+echo "  kill \$pid || exit 1" >>$config_dir/stop-init
+echo "  touch $var_dir/stop-flag" >>$config_dir/stop-init
+echo "  if [ \"\$1\" = \"--kill\" ]; then" >>$config_dir/stop-init
+echo "    sleep 1" >>$config_dir/stop-init
+echo "    ((ps axf | grep \"webmin\/miniserv\.pl\" | awk '{print \"kill -9 -- -\" \$1}' | bash) || kill -9 -- -\$pid || kill -9 \$pid) 2>/dev/null" >>$config_dir/stop-init
+echo "  fi" >>$config_dir/stop-init
+echo "  exit 0" >>$config_dir/stop-init
+echo "else" >>$config_dir/stop-init
+echo "  if [ \"\$1\" = \"--kill\" ]; then" >>$config_dir/stop-init
+echo "    (ps axf | grep \"webmin\/miniserv\.pl\" | awk '{print \"kill -9 -- -\" \$1}' | bash) 2>/dev/null" >>$config_dir/stop-init
+echo "  fi" >>$config_dir/stop-init
+echo "fi" >>$config_dir/stop-init
+# Restart main
+echo "#!/bin/sh" >>$config_dir/restart-init
+echo "$config_dir/stop-init" >>$config_dir/restart-init
+echo "$config_dir/start-init" >>$config_dir/restart-init
+# Force reload main
+echo "#!/bin/sh" >>$config_dir/force-reload-init
+echo "$config_dir/stop-init --kill" >>$config_dir/force-reload-init
+echo "$config_dir/start-init" >>$config_dir/force-reload-init
+# Reload main
+echo "#!/bin/sh" >>$config_dir/reload-init
+echo "echo Reloading Webmin server in $wadir" >>$config_dir/reload-init
+echo "pidfile=\`grep \"^pidfile=\" $config_dir/miniserv.conf | sed -e 's/pidfile=//g'\`" >>$config_dir/reload-init
+echo "kill -USR1 \`cat \$pidfile\`" >>$config_dir/reload-init
+
+chmod 755 $config_dir/stop-init $config_dir/start-init $config_dir/restart-init $config_dir/force-reload-init $config_dir/reload-init
 echo "..done"
 echo ""
 
+# Re-generating supplementary
+
+# Clear existing
+csupp () {
+	rm -f $config_dir/stop $config_dir/start $config_dir/restart $config_dir/force-reload $config_dir/reload
+}
+csupp
```

Comment:
Why does this need to be a sub? Also, is this syntax correct?

Reply:
The same as above, we call it twice.

This is Bash, why wouldn't syntax be incorrect?

Reply:
It's not safe to use bash-specific syntax - users might run setup.sh on systems like FreeBSD that have the older Bourne shell. Instead, just repeat this line.

Reply:
Okay, but if this is holding the merge and a release just go ahead and do it.

Reply:
I mean we couldn't do a release with this code as-is, as it would break non-Linux systems.

Reply:
Just run the `rm` statement directly where you're currently calling `csupp`